test/verify/check-machines-dbus

```diff
@@ -0,0 +1,67 @@
+#!/usr/bin/python3
+# -*- coding: utf-8 -*-
+
+# This file is part of Cockpit.
+#
+# Copyright (C) 2018 Red Hat, Inc.
+#
+# Cockpit is free software; you can redistribute it and/or modify it
+# under the terms of the GNU Lesser General Public License as published by
+# the Free Software Foundation; either version 2.1 of the License, or
+# (at your option) any later version.
+#
+# Cockpit is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+# Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with Cockpit; If not, see <http://www.gnu.org/licenses/>.
+
+import parent
+import unittest
+
+from testlib import *
+import machineslib
+
+
+class TestMachinesDBus(machineslib.TestMachines):
+
+    def detectLibvirtDbus(self):
+        m = self.machine
+        namesList = m.execute("dbus-send --system --dest=org.freedesktop.DBus "
+                              "--type=method_call "
+                              "--print-reply /org/freedesktop/DBus "
+                              "org.freedesktop.DBus.ListActivatableNames")
+
+        if ('org.libvirt' not in namesList):
+            raise unittest.SkipTest("libvirt-dbus is not available");
+        m.execute("echo 'PROVIDER=libvirt-dbus' > /etc/environment")
+
+    def allowSystemDBusUnprivileged(self):
+        m = self.machine
+        libvirtConfig = '<?xml version="1.0"?> \
+        <!DOCTYPE busconfig PUBLIC "-//freedesktop//DTD D-BUS Bus Configuration 1.0//EN" \
+        "http://www.freedesktop.org/standards/dbus/1.0/busconfig.dtd"> \
+        <busconfig> \
+        <policy user="admin"> \
+            <allow send_destination="org.libvirt"/> \
+        </policy> \
+        </busconfig>'
+        m.execute("echo '{0}' > {1}".format(libvirtConfig, "/etc/dbus-1/system.d/org.libvirt.conf"))
+
+
+    def setUp(self):
+        MachineCase.setUp(self)
+        self.startLibvirt()
+        self.detectLibvirtDbus()
+        self.allowSystemDBusUnprivileged()
+
+        # enforce use of cockpit-machines instead of cockpit-machines-ovirt
+        m = self.machine
+        m.execute("sed -i 's/\"priority\".*$/\"priority\": 100,/' {0}".format("/usr/share/cockpit/machines/manifest.json"))
+        m.execute("[ ! -e {0} ] || sed -i 's/\"priority\".*$/\"priority\": 0,/' {0}".format("/usr/share/cockpit/ovirt/manifest.json"))
```

Comment:
Doesn't that apply to the virsh test case as well? I. e. shuodln't that go/stay in machineslib.py?

Reply:
Actually this part of the setup function is the same in the two classes, I 'll move it to the parent class.

Reply:
Done